test/modules/layers/text-layer/text-layer.spec.js

```diff
@@ -54,6 +54,199 @@ test('TextLayer - sdf', t => {
   t.end();
 });
 
+test('TextLayer - MultiIconLayer sublayer positions and offsets', t => {
+  const getName = d => d.NAME;
+  const getEyeColor = d => d.EYE_COLOR;
+
+  const aliceCoordinates2d = [1, 2];
+  const aliceCoordinates3d = [...aliceCoordinates2d, 0];
+
+  const bobCoordinates2d = [3, 4];
+  const bobCoordinates3d = [...bobCoordinates2d, 0];
+
+  const emptyOffsets = [0, 0];
+  const emptyCoordinates3d = [0, 0, 0];
+
+  const testCases = [
+    {
+      props: {
+        data: [
+          {
+            NAME: 'Alice',
+            EYE_COLOR: 'blue',
+            COORDINATES: aliceCoordinates2d
+          },
+          {
+            NAME: 'Bob',
+            EYE_COLOR: 'brown',
+            COORDINATES: bobCoordinates2d
+          }
+        ],
+        getText: getName,
+        getPosition: d => d.COORDINATES,
+        updateTriggers: {
+          getText: [getName]
+        }
+      },
+      onAfterUpdate: ({subLayer}) => {
```

Comment:
The tests in this first `onAfterUpdate` are a sanity check of the MultiIconLayer attributes. I'd be happy to delete them if this PR is too verbose.